### ISSUE TITLE
Offline criteria did not create incident when web is unavailable

### ISSUE THREAD

**Issue by JanRajnoha** [OneUptime/oneuptime#703]
**Describe the bug**
Everytime when is my web unavailable, monitor will recognize correctly, that something is wrong, but incident is not created. Option to automatic resolve is unchecked.

**To Reproduce**
1. Set criteria to create incident when web is unavailable
2. Make web unavailable (not manually trigger criteria)
3. Incident is not created

**Expected behavior**
Incident is created and is not resolved automatically, when web is again online

**Screenshots**
![image](https://github.com/OneUptime/oneuptime/assets/16069721/ae6f8f50-21a0-493f-8ad7-8530b14efe58)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Edge
 - Version: Current Canary version

**Comment by simlarsen** [comment]
@JanRajnoha do you have the incident already active for that monitor? If so, we dont create another incident.

**Comment by JanRajnoha** [comment]
No, there is not any incident at all.

**Comment by simlarsen** [comment]
@JanRajnoha is this on a self hosted instance or SaaS?

**Comment by JanRajnoha** [comment]
It is SaaS and monitor ID is 4a7a3a97-ffc4-476c-9605-0ce276ce81c8

**Comment by simlarsen** [comment]
@JanRajnoha Thank you for this. Investigating. Will get back to you by EOD today.

**Comment by simlarsen** [comment]
@JanRajnoha You need to add Response Status Coide as a part of your criteria. You only have `Is Online` which will be true even if the server returns 500. As long as server responds to a request and does not timeout `Is Online` will be true.

**Comment by JanRajnoha** [comment]
I added check for status code != 200 and switch filters to Any, but nothing changed. Monitor is still switching to offline (as before), but incident is still not created

**Comment by simlarsen** [comment]
I dont think you added the response code filters. These are the only ones that show up: 

<img width="610" alt="image" src="https://github.com/OneUptime/oneuptime/assets/104437376/8c5298e9-a93b-47e6-a93b-f01e5bdb0d82">

**Comment by JanRajnoha** [comment]
I'm sorry, but you did not read my previous message.

> I added check for status code != 200 and switch filters to Any, but nothing changed. Monitor is still switching to offline (as before), but incident is still not created

Or you want to say, that monitor will trigger (in current state) offline status, but not create incident? Monitor is working as should, but incidents are not created. I see in my monitor, that service is down like I should, but nothing else.

I'm simulating, that service is down currently. In one hour probably results will be visible in monitor, you could check AGAIN that nothing will happen.

**Comment by JanRajnoha** [comment]
Current state:
![image](https://github.com/OneUptime/oneuptime/assets/16069721/881f77f1-e9da-4c9c-ae95-01035126ec75)
![image](https://github.com/OneUptime/oneuptime/assets/16069721/89ed2079-7a62-451b-980b-537420adf98d)

**Comment by simlarsen** [comment]
@JanRajnoha Can you please also share screenshot of criteria page?

**Comment by JanRajnoha** [comment]
![image](https://github.com/OneUptime/oneuptime/assets/16069721/dedb4108-4855-4daa-8c52-4beab3694ca0)

**Comment by JanRajnoha** [comment]
It is working for offline too (changing status), but incident don't create

**Comment by JanRajnoha** [comment]
I tried different criteria:
![image](https://github.com/OneUptime/oneuptime/assets/16069721/3d3a663a-2fc5-42ae-a907-ba038ce205f5)
Same result -> status changed correctly, but incident has not been created

**Comment by sresupport** [comment]
Hi @JanRajnoha / @simlarsen any update on this issue? I also face this issue on Self hosted version 10.0.17. 

As a workaround I happened to change the order of the criteria, means I have added online checks as first criteria and the offline checks as second criteria. But we have a lot of monitors running on OneUptime, this solution is not feasible.

**Comment by simlarsen** [comment]
@sresupport / @JanRajnoha This has been fixed a long time ago. Please upgrade to latest. Please check monitor logs to see why the incident wasnt created.
